src/Repository/PasswordResetTokenRepository.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace BigGive\Identity\Repository;
+
+use BigGive\Identity\Domain\PasswordResetToken;
+use Doctrine\ORM\EntityRepository;
+use Symfony\Component\Uid\Uuid;
+
+class PasswordResetTokenRepository extends EntityRepository
+{
+    public function persist(PasswordResetToken $token): void
+    {
+        $this->getEntityManager()->persist($token);
+        $this->getEntityManager()->flush();
+    }
+
+    public function findForUse(Uuid $secret): ?PasswordResetToken
+    {
+        $query = $this->_em->createQuery(
+            "SELECT p from \BigGive\Identity\Domain\PasswordResetToken p
+            WHERE p.secret = :secret
+            AND p.used IS NULL
+            AND p.created_at > DATE_SUB(CURRENT_TIMESTAMP(), 1, 'HOUR') 
+            "
+        );
+        $query->setParameter('secret', $secret->toBinary());
```

Comment:
@bdsl Would it be worth explicitly binding with `Types::BINARY` in the 3rd arg so we know for sure that what we're binding is what the ORM expects? I think it might also automatically convert if necessary though haven't tried with a binary field before.

Reply:
Yes, done. Probably worth always passing a 3rd param to setParameter.